lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -892,13 +897,17 @@ fn generate_constraint_associated_token(
     quote! {
         {
             #optional_checks
-
+            let mint_program_id = {
+              let mint_account = #spl_token_mint_address.to_account_info();
```

Comment:
Same as above, `to_account_info()` internally uses `clone()` to get the `AccountInfo`. We can avoid cloning by using:

```suggestion
              let mint_account: &AccountInfo = #spl_token_mint_address.as_ref();
```